x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -154,6 +163,15 @@ func (k Keeper) CheckDelegationChangedDuringQuery(
 		}
 	}
 
+	// Check that there are no LSMTokenDeposits in state IN_PROGRESS - indictive of an Detokenization ICA
+	for _, lsmTokenDeposit := range k.RecordsKeeper.GetLSMDepositsForHostZone(ctx, chainId) {
```

Comment:
Use GetLSMDepositsForHostZoneWithStatus because then if there are really none in progress, you loop 0 times

Reply:
ah good call, forgot about that one!